### PR TITLE
Add tests for various scenarios for a faulty bjobs

### LINKED ISSUE #7186: Harden Python LSF driver towards faulty bsub/bjobs/bkill
* Write tests that can mock faulty return codes and garbage output on stdout/stderr from the LSF commands bsub, bjobs, bkill
* Assert that ERT will handle these errors in some way. Identify scenarios that ERT should recognize and do a retry, error hard when relevant.
* Fall back to bhist: #7208

### PR DESCRIPTION
**Pull request by berland**
**Issue**
Partly resolved #7186 

**Approach**
Add tests certifying the current behaviour of `poll()`. Not necessarily the future behaviour.

- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: tests/integration_tests/scheduler/test_lsf_driver.py [hunk 1]
```diff
@@ -152,3 +156,79 @@ async def finished(iens, returncode, aborted):
...
+            id="correct_output_but_exitcode_1",
+        ),
+        pytest.param(
+            f"echo '{BJOBS_HEADER}\n1 someuser'; exit 0",
+            pytest.raises(asyncio.TimeoutError),
+            id="unparsable_output",
+        ),
+    ],
+)
+async def test_faulty_bjobs(monkeypatch, tmp_path, bjobs_script, expectation):
```
Comment (xjules) [inline tests/integration_tests/scheduler/test_lsf_driver.py:220]: Why is this an integration test?

File: tests/integration_tests/scheduler/test_lsf_driver.py [hunk 2]
```diff
@@ -152,3 +156,79 @@ async def finished(iens, returncode, aborted):
...
+    bin_path.mkdir()
+    monkeypatch.setenv("PATH", f"{bin_path}:{os.environ['PATH']}")
+    bsub_path = bin_path / "bsub"
+    bsub_path.write_text("#!/bin/sh\necho 'Job <1> is submitted to default queue'")
+    bsub_path.chmod(bsub_path.stat().st_mode | stat.S_IEXEC)
+    bjobs_path = bin_path / "bjobs"
+    bjobs_path.write_text(f"#!/bin/sh\n{bjobs_script}")
+    bjobs_path.chmod(bsub_path.stat().st_mode | stat.S_IEXEC)
+    driver = LsfDriver()
+    driver._poll_period = 0.01
```
Comment (xjules) [inline tests/integration_tests/scheduler/test_lsf_driver.py:231]: Does it even matter to set the `driver._poll_period` here?

File: src/ert/scheduler/lsf_driver.py [hunk 3]
```diff
@@ -146,17 +156,21 @@ async def kill(self, iens: int) -> None:
...
                 stderr=asyncio.subprocess.PIPE,
             )
 
-            stdout, _ = await proc.communicate()
-            stat = _Stat(**parse_bjobs(stdout))
+            stdout, stderr = await process.communicate()
+            if process.returncode:
+                logger.error(
+                    f"bjobs gave returncode {process.returncode} and error {stderr.decode()}"
+                )
```
Comment (xjules) [inline src/ert/scheduler/lsf_driver.py:172]: `continue` ?

File: tests/integration_tests/scheduler/test_lsf_driver.py (hunk 1 above)
Comment (berland) [reply tests/integration_tests/scheduler/test_lsf_driver.py:220]: It is not. Moved.

File: tests/integration_tests/scheduler/test_lsf_driver.py (hunk 2 above)
Comment (berland) [reply tests/integration_tests/scheduler/test_lsf_driver.py:231]: Nope, does not make it faster. Removed.

File: src/ert/scheduler/lsf_driver.py (hunk 3 above)
Comment (berland) [reply src/ert/scheduler/lsf_driver.py:172]: yeah, that makes sense!

Comment (berland) [reply src/ert/scheduler/lsf_driver.py:203]: No it does not make sens, as there is a test that will fail if we add continue here. I don't trust the return code from bjobs, see the test where it returns 255.

Comment (berland) [reply src/ert/scheduler/lsf_driver.py:203]: downgrading loglevel to warning.

File: src/ert/scheduler/lsf_driver.py [hunk 4]
```diff
@@ -56,13 +57,20 @@ class _Stat(BaseModel):
     jobs: Mapping[str, AnyJob]
 
 
-def parse_bjobs(bjobs_output_raw: bytes) -> Dict[str, Dict[str, Dict[str, str]]]:
+def parse_bjobs(bjobs_output: str) -> Dict[str, Dict[str, Dict[str, str]]]:
     data: Dict[str, Dict[str, str]] = {}
-    for line in bjobs_output_raw.decode().splitlines():
+    for line in bjobs_output.splitlines():
```
Comment (xjules) [inline src/ert/scheduler/lsf_driver.py:62]: so there could be an empty line then?

**Review by xjules — APPROVED** [review]
I think it looks good now! :rocket:

**Pull request merged**